Networking/Networking/Model/Blaze/CreateBlazeCampaign.swift

```diff
@@ -118,4 +118,56 @@ public struct CreateBlazeCampaign: Encodable, GeneratedFakeable, GeneratedCopiab
         self.targetUrn = targetUrn
         self.type = type
     }
+
+    public init(origin: String,
+                originVersion: String,
+                paymentMethodID: String,
+                startDate: Date,
+                endDate: Date,
+                timeZone: String,
+                totalBudget: Double,
+                siteName: String,
+                textSnippet: String,
+                destinationURL: String,
+                mainImage: Image,
+                targeting: BlazeTargetOptions?,
+                targetUrn: String,
+                type: String) {
+        let (targetUrl, urlParams) = destinationURL.targetURLAndParams()
+        self.init(origin: origin,
+                  originVersion: originVersion,
+                  paymentMethodID: paymentMethodID,
+                  startDate: startDate,
+                  endDate: endDate,
+                  timeZone: timeZone,
+                  totalBudget: totalBudget,
+                  siteName: siteName,
+                  textSnippet: textSnippet,
+                  targetUrl: targetUrl,
+                  urlParams: urlParams,
+                  mainImage: mainImage,
+                  targeting: targeting,
+                  targetUrn: targetUrn,
+                  type: type)
+    }
+}
+
+// MARK: Helper to fetch URL and params
+private extension String {
+    func targetURLAndParams() -> (targetUrl: String, urlParams: String) {
```

Comment:
This works, but I wonder why we don't just send back the target URL and params from `BlazeAdDestinationSettingViewModel` to the campaign creation form view model to avoid having to do this calculation again?